src/main/api/opds.ts

```diff
@@ -65,8 +65,13 @@ export class OpdsApi {
     }
 
     public async browse(data: any): Promise<THttpGetOpdsResultView> {
-        const { url } = data;
-        return await httpGet(url, {}, async (opdsFeedData) => {
+        let url: string = data.url;
+        if (new URL(url).protocol === "opds") {
+            url = url.replace(/([a-zA-Z]*)\:\/\//, "http");
```

Comment:
This regular expression does not seem to work. What URL did you successfully test it with?

To reproduce the bug, copy-paste this Javascript code snippet in your web browser console:
`'opds://127.0.0.1:8080/test.atom'.replace(/([a-zA-Z]*)\:\/\//, "http")`
=>
`"http127.0.0.1:8080/test.atom"`

Several issues:
1) The capturing group `([a-zA-Z]*)` is not consumed anywhere in the replacement pattern (i.e. `$1`).
2) The sequence `\:` escapes a non-reserved character. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
3) The `[a-zA-Z]*` quantifier should be `+`, not `*`.
4) The `[a-zA-Z]` character definition does not match the URL scheme syntax https://tools.ietf.org/html/rfc3986?#section-3.1 (`ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )`)
5) To be precise, the boundary "beginning" indicator `^` should be specified in the `RegExp`, before the URL scheme matcher (caveat: `url` *must* be trimmed from whitespace characters on the leading edge).

To conclude, because the condition `new URL(url).protocol === "opds"` is `true`, personally I would avoid using regular expressions, and I would simply replace `opds://` with `http://`, e.g. `'opds://127.0.0.1:8080/test.atom'.replace("opds://", "http://")` => `http://127.0.0.1:8080/test.atom`

Reply:
many thanks Daniel for your explanation

I fixed it